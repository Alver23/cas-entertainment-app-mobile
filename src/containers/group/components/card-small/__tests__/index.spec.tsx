// Dependencies
import React from 'react';
import { Text } from 'react-native-elements';
import { shallow, ShallowWrapper } from 'enzyme';

// Components
import CardSmall from '../index';

// Mocks
import mocks from './mocks.json';

describe('<CardSmall />', () => {
  let component: ShallowWrapper;

  const props = { ...mocks };

  beforeEach(() => {
    component = shallow(<CardSmall {...props} />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should save a snapshot of the component', () => {
    expect(component).toMatchSnapshot();
    expect(component.find(Text)).toHaveLength(2);
  });

  it('should render correctly when state is not null', () => {
    component.setProps({ ...mocks, state: 'active' });
    expect(component.find(Text)).toHaveLength(3);
  });
});
