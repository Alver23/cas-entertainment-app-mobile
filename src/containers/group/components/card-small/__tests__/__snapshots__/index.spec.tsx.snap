// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardSmall /> should save a snapshot of the component 1`] = `
<View
  style={
    Object {
      "justifyContent": "space-between",
      "width": 330,
    }
  }
>
  <Card
    status="green"
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "column",
            "width": "auto",
          }
        }
      >
        <Themed.Text
          numberOfLines={1}
          style={
            Object {
              "fontSize": 26.5,
            }
          }
        >
          fake title
        </Themed.Text>
        <Themed.Text
          style={
            Object {
              "color": undefined,
              "fontSize": 21.5,
            }
          }
        >
          (
          fake description
          )
        </Themed.Text>
      </View>
    </View>
  </Card>
</View>
`;
